tensorboard/components/vz_line_chart/dragZoomInteraction.ts

```diff
@@ -1,227 +1,217 @@
-     * d3.ease('cubic-in-out')) and is customizable.
-     * Usage: Construct the selection box layer and attach x and y scales,
-     * and then add the layer over the plot you are zooming on using a
-     * Component Group.
-     * TODO(@decentralion) - merge this into Plottable
-     */
-    constructor(
-      xScale: Plottable.QuantitativeScale<number | {valueOf(): number}>,
-      yScale: Plottable.QuantitativeScale<number | {valueOf(): number}>,
-      unzoomMethod: Function
-    ) {
-      super();
-      this.xScale(xScale);
-      this.yScale(yScale);
-      this._dragInteraction = new Plottable.Interactions.Drag();
-      this._doubleClickInteraction = new Plottable.Interactions.Click();
-      this.setupCallbacks();
-      this.unzoomMethod = unzoomMethod;
-
-      // Activate interaction only when the component is mounted onto DOM.
-      this.onDetach(() => {
-        this._doubleClickInteraction.detachFrom(this);
-        this._dragInteraction.detachFrom(this);
-      });
-      this.onAnchor(() => {
-        this._doubleClickInteraction.attachTo(this);
-        this._dragInteraction.attachTo(this);
-      });
-    }
-
-    /**
-     * Register a method that calls when the DragZoom interaction starts.
-     */
-    public interactionStart(cb: Function) {
-      this.onStart = cb;
-    }
-
-    /**
-     * Register a method that calls when the DragZoom interaction ends.
-     */
-    public interactionEnd(cb: Function) {
-      this.onEnd = cb;
-    }
-
-    /**
-     * Returns backing drag interaction. Useful for customization to the
-     * interaction.
-     */
-    public dragInteraction(): Plottable.Interactions.Drag {
-      return this._dragInteraction;
-    }
-
-    private setupCallbacks() {
-      let dragging = false;
-      this._dragInteraction.onDragStart((startPoint: Plottable.Point) => {
-        this.bounds({
-          topLeft: startPoint,
-          bottomRight: startPoint,
-        });
-        this.onStart();
-      });
-      this._dragInteraction.onDrag((startPoint, endPoint) => {
-        this.bounds({topLeft: startPoint, bottomRight: endPoint});
-        this.boxVisible(true);
-        dragging = true;
-      });
-      this._dragInteraction.onDragEnd((startPoint, endPoint) => {
-        this.boxVisible(false);
-        this.bounds({topLeft: startPoint, bottomRight: endPoint});
-        if (dragging) {
-          this.zoom();
-        } else {
-          this.onEnd();
+import * as d3 from 'd3';
+import * as Plottable from 'plottable';
+
+export class DragZoomLayer extends Plottable.Components.SelectionBoxLayer {
+  private _dragInteraction: Plottable.Interactions.Drag;
+  private _doubleClickInteraction: Plottable.Interactions.Click;
+  private easeFn: (t: number) => number = d3.easeCubicInOut;
+  private _animationTime = 750;
+  private onStart: () => void = () => void {};
+  private onEnd: () => void = () => void {};
+  private unzoomMethod: () => void;
```

Comment:
Doesn't unzoomMethod also need to be initialized?

Reply:
unzoomMethod is initialized in the constructor.

Reply:
Ahh sorry, I didn't properly read!